src/stream/src/executor/dynamic_filter.rs

```diff
@@ -415,6 +440,10 @@ impl<S: StateStore, const USE_WATERMARK_CACHE: bool> DynamicFilterExecutor<S, US
                         if let Some(chunk) = stream_chunk_builder.take() {
                             yield Message::Chunk(chunk);
                         }
+
+                        for r in to_delete_rows {
+                            self.left_table.delete(r);
```

Comment:
Why not deleting them by `range_delete`?

Reply:
LGTM, but I will take a think later :thinking:

Reply:
Either LGTM. `range_delete` sounds a bit more intuitive to me.

Reply:
Yes, I think we can use `range_delete`. I got it wrong because I used to consider "always relax" more complex, but It actually a lower boundary on the table because of the dynamic filter's property.